src/coreclr/jit/importercalls.cpp

```diff
@@ -1288,7 +1288,7 @@ var_types Compiler::impImportCall(OPCODE                  opcode,
         //    have to check for anything that might introduce a recursive tail call.
         // * We only instrument root method blocks in OSR methods,
         //
-        if (opts.IsOSR() && !compIsForInlining())
+        if ((opts.IsInstrumentedOptimized() || opts.IsOSR()) && !compIsForInlining())
```

Comment:
As noted below, I think this is an OSR only problem.

Reply:
So it could be 
```suggestion
        if ((opts.IsInstrumentedOptimized() && opts.IsOSR()) && !compIsForInlining())
```

Reply:
`opts.IsInstrumentedOptimized() && opts.IsOSR()` is not working, for some reason it hits an assert even in non PGO OSR mode. I should be able to revert these changes once I fix "edge profiling" for optimized code